Donation-platform/src/main/java/com/google/FindTrendingCharities.java

```diff
@@ -30,144 +30,150 @@
 
 public final class FindTrendingCharities {
 
-  private DbCalls db;
-  private DatastoreService ds;
-
-  // number of trending charities to be returned
-  final int MAX_NUM_OF_CHARITIES_TO_RETURN = 7;
-
-  final double CHARITY_NAV_SCALE_FACTOR = 1.25;
-
-  // should sum to 1
-  final double USER_RATING_WEIGHT = 0.75;
-  final double CHARITY_NAV_WEIGHT = 0.25;
-
-  // should sum to 1
-  final double TAGS_SCORE_WEIGHT = 0.75;
-  final double AVG_REVIEW_WEIGHT = 0.25;
-
-  private static Collection<Charity> charities;
-
-  //constructor to do set up
-  public FindTrendingCharities(DatastoreService ds) {
-    //ds = DatastoreServiceFactory.getDatastoreService();
-    this.ds = ds;
-    db = new DbCalls(ds);
-    DbSetUpUtils setUp = new DbSetUpUtils(ds, db);
-    this.charities = getAllCharities();
-    // only populate database if there is nothing in the database already
-    if (charities.size() < 1) {
-        setUp.populateDatabase();
-    }
-  }
-
-  // returns the collection of top trending charities
-  public Collection<Charity> queryDb() {
-    //Collection<Charity> charities = getAllCharities();
-    for (Charity charity : charities) {
-      double charityScore = calcCharityTrendingScore(charity);
-      charity.setTrendingScoreCharity(charityScore);
-      try {
-          db.updateCharity(charity);
-      }
-      catch (Exception e) {
-          System.out.println("unable to update charity: " + e);
-      }
+    private DbCalls db;
+    private DatastoreService ds;
+
+    // number of trending charities to be returned
+    final int MAX_NUM_OF_CHARITIES_TO_RETURN = 10;
+
+    final double CHARITY_NAV_SCALE_FACTOR = 1.25;
+
+    // should sum to 1
+    final double USER_RATING_WEIGHT = 0.75;
+    final double CHARITY_NAV_WEIGHT = 0.25;
+
+    // should sum to 1
+    final double TAGS_SCORE_WEIGHT = 0.75;
+    final double AVG_REVIEW_WEIGHT = 0.25;
+
+    private static Collection<Charity> charities;
+    private static List<Charity> topTrendingCharities;
+
+    //constructor to do set up
+    public FindTrendingCharities(DatastoreService ds) {
+        //ds = DatastoreServiceFactory.getDatastoreService();
+        this.ds = ds;
+        db = new DbCalls(ds);
+        DbSetUpUtils setUp = new DbSetUpUtils(ds, db);
+        this.charities = getAllCharities();
+        this.topTrendingCharities = new ArrayList<Charity>();
+        // only populate database if there is nothing in the database already
+        if (charities.size() < 1) {
```

Comment:
This check is now done in personalization. My upcoming PR removes this if statement block.

Reply:
Is it possible for the database to populate outside of any of our functions. I feel as if this sort of nesting prevents us from abstracting such a function and making it dependent on a particular page doesn't seem efficient. Could we not have a function load up the database on website load, regardless of page or functions called?

Reply:
Yeah I agree that's a better solution long-term. We can talk about the best way to do that.

Reply:
i agree with Toni's point. we should have an abstract class that takes care of connecting to db etc and avoid making db connection at multiple places. This might also lead to db inconsistency